<h2>Задача 1</h2>
Скачать файл в уроке<br>
Считать данные с помощью pandas<br>
Вывести на экран первые 5 строк<br>
Посмотреть на описание признаков и на их содержание


In [1]:
import pandas as pd

a = pd.read_csv('./HomeWork_2/kc_final (1).csv')
df = pd.DataFrame(a)
df.head(5)

Unnamed: 0          id        date     price  bedrooms  bathrooms  \
0           0  7129300520  2014-10-13  221900.0         3       1.00   
1           1  6414100192  2014-12-09  538000.0         3       2.25   
2           2  5631500400  2015-02-25  180000.0         2       1.00   
3           3  2487200875  2014-12-09  604000.0         4       3.00   
4           4  1954400510  2015-02-18  510000.0         3       2.00   

   sqft_living  sqft_lot  floors  waterfront  ...  grade  sqft_above  \
0         1180      5650     1.0           0  ...      7        1180   
1         2570      7242     2.0           0  ...      7        2170   
2          770     10000     1.0           0  ...      6         770   
3         1960      5000     1.0           0  ...      7        1050   
4         1680      8080     1.0           0  ...      8        1680   

   sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
0              0      1955             0    98178  47.5112 -122.257   
1            400      1951          1991    98125  47.7210 -122.319   
2              0      1933             0    98028  47.7379 -122.233   
3            910      1965             0    98136  47.5208 -122.393   
4              0      1987             0    98074  47.6168 -122.045   

   sqft_living15  sqft_lot15  
0           1340        5650  
1           1690        7639  
2           2720        8062  
3           1360        5000  
4           1800        7503  

[5 rows x 22 columns]

In [2]:
df.describe()

Unnamed: 0            id         price      bedrooms     bathrooms  \
count  21613.00000  2.161300e+04  2.161300e+04  21613.000000  21613.000000   
mean   10806.00000  4.580302e+09  5.400881e+05      3.370842      2.114757   
std     6239.28002  2.876566e+09  3.671272e+05      0.930062      0.770163   
min        0.00000  1.000102e+06  7.500000e+04      0.000000      0.000000   
25%     5403.00000  2.123049e+09  3.219500e+05      3.000000      1.750000   
50%    10806.00000  3.904930e+09  4.500000e+05      3.000000      2.250000   
75%    16209.00000  7.308900e+09  6.450000e+05      4.000000      2.500000   
max    21612.00000  9.900000e+09  7.700000e+06     33.000000      8.000000   

        sqft_living      sqft_lot        floors    waterfront          view  \
count  21613.000000  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean    2079.899736  1.510697e+04      1.494309      0.007542      0.234303   
std      918.440897  4.142051e+04      0.539989      0.086517      0.766318   
min      290.000000  5.200000e+02      1.000000      0.000000      0.000000   
25%     1427.000000  5.040000e+03      1.000000      0.000000      0.000000   
50%     1910.000000  7.618000e+03      1.500000      0.000000      0.000000   
75%     2550.000000  1.068800e+04      2.000000      0.000000      0.000000   
max    13540.000000  1.651359e+06      3.500000      1.000000      4.000000   

       ...         grade    sqft_above  sqft_basement      yr_built  \
count  ...  21613.000000  21613.000000   21613.000000  21613.000000   
mean   ...      7.656873   1788.390691     291.509045   1971.005136   
std    ...      1.175459    828.090978     442.575043     29.373411   
min    ...      1.000000    290.000000       0.000000   1900.000000   
25%    ...      7.000000   1190.000000       0.000000   1951.000000   
50%    ...      7.000000   1560.000000       0.000000   1975.000000   
75%    ...      8.000000   2210.000000     560.000000   1997.000000   
max    ...     13.000000   9410.000000    4820.000000   2015.000000   

       yr_renovated       zipcode           lat          long  sqft_living15  \
count  21613.000000  21613.000000  21613.000000  21613.000000   21613.000000   
mean      84.402258  98077.939805     47.560053   -122.213896    1986.552492   
std      401.679240     53.505026      0.138564      0.140828     685.391304   
min        0.000000  98001.000000     47.155900   -122.519000     399.000000   
25%        0.000000  98033.000000     47.471000   -122.328000    1490.000000   
50%        0.000000  98065.000000     47.571800   -122.230000    1840.000000   
75%        0.000000  98118.000000     47.678000   -122.125000    2360.000000   
max     2015.000000  98199.000000     47.777600   -121.315000    6210.000000   

          sqft_lot15  
count   21613.000000  
mean    12768.455652  
std     27304.179631  
min       651.000000  
25%      5100.000000  
50%      7620.000000  
75%     10083.000000  
max    871200.000000  

[8 rows x 21 columns]

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 22 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     21613 non-null  int64  
 1   id             21613 non-null  int64  
 2   date           21613 non-null  object 
 3   price          21613 non-null  float64
 4   bedrooms       21613 non-null  int64  
 5   bathrooms      21613 non-null  float64
 6   sqft_living    21613 non-null  int64  
 7   sqft_lot       21613 non-null  int64  
 8   floors         21613 non-null  float64
 9   waterfront     21613 non-null  int64  
 10  view           21613 non-null  int64  
 11  condition      21613 non-null  int64  
 12  grade          21613 non-null  int64  
 13  sqft_above     21613 non-null  int64  
 14  sqft_basement  21613 non-null  int64  
 15  yr_built       21613 non-null  int64  
 16  yr_renovated   21613 non-null  int64  
 17  zipcode        21613 non-null  int64  
 18  lat   

In [6]:
df.isnull().sum()

Unnamed: 0       0
id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

In [22]:
df.loc[df['waterfront'] == 1]

Unnamed: 0          id        date      price  bedrooms  bathrooms  \
49             49   822039084  2015-03-11  1350000.0         3       2.50   
230           230  8096000060  2015-04-13   655000.0         2       1.75   
246           246  2025069065  2014-09-29  2400000.0         4       2.50   
264           264  2123039032  2014-10-27   369900.0         1       0.75   
300           300  3225069065  2014-06-24  3075000.0         4       5.00   
...           ...         ...         ...        ...       ...        ...   
19984       19984  2025069140  2015-03-17  1898000.0         3       2.50   
20325       20325   518500480  2014-08-11  3000000.0         3       3.50   
20767       20767  8043700105  2015-04-17  2300000.0         4       4.00   
21201       21201   518500460  2014-10-08  2230000.0         3       3.50   
21576       21576  9253900271  2015-01-07  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  ...  grade  sqft_above  \
49            2753     65005     1.0           1  ...      9        2165   
230           1450     15798     2.0           1  ...      7        1230   
246           3650      8354     1.0           1  ...      9        1830   
264            760     10079     1.0           1  ...      5         760   
300           4550     18641     1.0           1  ...     10        2600   
...            ...       ...     ...         ...  ...    ...         ...   
19984         2830      4334     3.0           1  ...     10        2830   
20325         4410     10756     2.0           1  ...     11        3430   
20767         4360      8175     2.5           1  ...     10        3940   
21201         3760      5634     2.0           1  ...     11        2830   
21576         4850     10584     2.0           1  ...     10        3540   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
49               588      1953             0    98070  47.4041 -122.451   
230              220      1915          1978    98166  47.4497 -122.375   
246             1820      2000             0    98074  47.6338 -122.072   
264                0      1936             0    98070  47.4683 -122.438   
300             1950      2002             0    98074  47.6053 -122.077   
...              ...       ...           ...      ...      ...      ...   
19984              0      2006             0    98074  47.6318 -122.071   
20325            980      2014             0    98056  47.5283 -122.205   
20767            420      2007             0    98008  47.5724 -122.104   
21201            930      2014             0    98056  47.5285 -122.205   
21576           1310      2007             0    98008  47.5943 -122.110   

       sqft_living15  sqft_lot15  
49              2680       72513  
230             2030       13193  
246             3120       18841  
264             1230       14267  
300             4550       19508  
...              ...         ...  
19984           2830       38211  
20325           3550        5634  
20767           2670        8525  
21201           3560        5762  
21576           3470       18270  

[163 rows x 22 columns]

У нас имеется база данных с жилыми участками на 21613 записей.
В таблице нет пустых ячеек.
В таблице указаны: Жилая площадь, площадь участков, количество этажей, почтовый индекс, находится ли участок на береговой линии, год постройки участка, оценка участка, цена участка и т.д.
75% домов построено до 1997 года.

<h2>Задача 3</h2>
Ответьте на несколько вопросов<br>
3.1 В каком диапазоне изменяются стоимости недвижимости?<br>
3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?<br>
3.3 Как много домов с разными этажами в данных?<br>
3.4 Насколько хорошие состояния у домов в данных?<br>
3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?<br>

3.1 Диапозон цен 75000$ - 7700000$<br>
3.2 Примерно 13,77% занимает жилая площадь<br>
3.3 <li>1 этажный - 10680</li>
    <li>2 этажный - 8241</li>
    <li>3 этажный - 613</li>
    <li>1,5 этажа - 1910</li>
    <li>2,5 этажа - 161</li>
    <li>3,5 этажа - 8</li>
    <br>
3.4 Средняя оценка домов 7,65<br>
3.5 Первый дом был построе - 1900 год, последний дом был построен - 2015 год

In [32]:
df['floors'].value_counts()

floors
1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: count, dtype: int64

<h2>Задача 4</h2>
Ответьте на несколько вопросов<br>
<br>
4.1 Сколько в среднем стоят дома, у которых 2 спальни?<br>
4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?<br>
4.3 Как много домов коснулся ремонт?<br>
4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?<br>

In [55]:
df['price'][df['bedrooms'] == 2].mean()

401372.681884058

<h3>4.1</h3>
Средняя стоимость таких домов 401372,7$

In [58]:
print(df['sqft_living'][df['price'] >= 600000].mean())
print(df['sqft_lot'][df['price'] >= 600000].mean())


2885.329563305058
20454.783537543197


<h3>4.2</h3>
Средняя жилая площадь таких домов - 2885,33 квадратных футов
Средняя площадь участка - 20454,78 квадратных фута

In [40]:
df[df['yr_renovated']!= 0].value_counts().sum()

914

<h3>4.3</h3>
914 домов подверглись ремонту

In [52]:
grade10 = df[df['grade'] >= 10]
grade4 = df[df['grade'] <= 4]
razn = grade10['price'].mean() - grade4['price'].mean()
razn

1046589.4030340074

<h3>4.4</h3>
В среднем такие дома различаются по стоимости на 1046589,4$

<h2>Задача 5</h2>
Ответьте на несколько вопросов<br>
5.1 Выберите дом клиенту<br>
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?<br>
5.2 Выберите дом клиенту<br>
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?<br>
5.3 Выберите дом клиенту<br>
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [65]:
df[(df['waterfront']==1) & (df['bathrooms']>=3) & (df['sqft_basement']>0)].shape[0]

41

<h3>5.1</h3>
41 вариант

In [12]:
minPrice = df['price'][(df['view'] == 1) | (df['waterfront'] == 1) & (df['grade'] >= 12) & (df['yr_built']>=1980)].min()
maxPrice = df['price'][(df['view']) | (df['waterfront']) & (df['grade'] >= 12) & (df['yr_built']>=1980)].max()
print(f'Ценовой диапозон {minPrice}$ - {maxPrice}$')

Ценовой диапозон 217000.0$ - 7700000.0$


In [13]:
df['grade'][(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price']<=150000)].mean()

7.0

<h3>5.3</h3>
Средняя оценка таких домов 7.0